File: utils/src/main/java/love/forte/common/collections/LRULinkedHashMap.java
package love.forte.common.collections;

import java.util.LinkedHashMap;
import java.util.Map;

/**
 * @author ForteScarlet
 */
public class LRULinkedHashMap<K, V> extends LinkedHashMap<K, V> {

    private volatile int capacity;

    public int getCapacity() {
        return capacity;
    }

    public void setCapacity(int capacity) {
        this.capacity = capacity;
    }

    public LRULinkedHashMap(int initialCapacity, float loadFactor, int capacity) {
        super(initialCapacity, loadFactor);
        this.capacity = capacity;
    }

    public LRULinkedHashMap(int initialCapacity, int capacity) {
        super(initialCapacity);
        this.capacity = capacity;
    }

    public LRULinkedHashMap(int capacity) {
        this.capacity = capacity;
    }

    public LRULinkedHashMap(Map<? extends K, ? extends V> m, int capacity) {
        super(m);
        this.capacity = capacity;
    }

    public LRULinkedHashMap(int initialCapacity, float loadFactor, boolean accessOrder, int capacity) {
        super(initialCapacity, loadFactor, accessOrder);
        this.capacity = capacity;
    }

    @Override
    protected boolean removeEldestEntry(Map.Entry<K, V> eldest) {
        return size() > capacity;
    }
}

File: utils/src/main/java/love/forte/common/collections/LRULinkedHashMap.kt
package love.forte.common.collections


/**
 *
 * Lru map 最基础的实现。
 *
 */
@Deprecated("Use LRULinkedHashMap")
public class _LRULinkedHashMap<K, V> : LinkedHashMap<K, V> {
    @Volatile
    var capacity: Int

    constructor(capacity: Int, initialCapacity: Int, loadFactor: Float) : super(initialCapacity, loadFactor, true) {
        this.capacity = capacity
    }

    constructor(capacity: Int, initialCapacity: Int) : super(initialCapacity, 0.75f, true) {
        this.capacity = capacity
    }

    constructor(capacity: Int) : super() {
        this.capacity = capacity
    }

    constructor(capacity: Int, m: MutableMap<out K, out V>?) : super(m) {
        this.capacity = capacity
    }

    constructor(capacity: Int, initialCapacity: Int, loadFactor: Float, accessOrder: Boolean) : super(
        initialCapacity,
        loadFactor,
        accessOrder
    ) {
        this.capacity = capacity
    }

    override fun removeEldestEntry(eldest: MutableMap.MutableEntry<K, V>?): Boolean {
        return size > capacity
    }

}